tutor/src/components/change-student-id.cjsx

```diff
@@ -0,0 +1,104 @@
+_ = require 'lodash'
+React = require 'react'
+BS = require 'react-bootstrap'
+
+LoadableItem = require './loadable-item'
+{ ChangeStudentIdForm } = require 'shared'
+{ TransitionActions, TransitionStore } = require '../flux/transition'
+{ CourseStore, CourseActions } = require '../flux/course'
+{ StudentIdStore, StudentIdActions } = require '../flux/student-id'
+
+ERROR_MAP = {
+  student_identifier_has_already_been_taken:
+    'The provided student ID has already been used in this course. ' +
+    'Please try again or contact your instructor.'
+  blank_student_identifer:
+    'The student ID field cannot be left blank. Please enter your student ID.'
+}
+
+module.exports = React.createClass
+  contextTypes:
+    router: React.PropTypes.func
+
+  onCancel: -> @goBack()
+
+  componentWillMount: ->
+    {courseId} = @context.router.getCurrentParams()
+    CourseStore.on('course.loaded', @update)
+    CourseActions.load(courseId)
+
+  componentWillUnmount: ->
+    StudentIdStore.off('student-id-saved', @saved)
+    StudentIdStore.off('student-id-error', @update)
+
+  update: ->
+    @setState({})
+
+  goBack: ->
+    historyInfo = TransitionStore.getPrevious(@context.router)
+    url = historyInfo.path or 'dashboard'
+    @context.router.transitionTo(url)
+
+  onSubmit: (newStudentId) ->
+    {courseId} = @context.router.getCurrentParams()
+
+    if not newStudentId
+      StudentIdActions.addError(courseId, code:'blank_student_identifer')
+      @setState({})
+      return
+
+    StudentIdStore.on('student-id-saved', @saved)
+    StudentIdStore.on('student-id-error', @update)
+    StudentIdActions.save(courseId, {
```

Comment:
I think using explicit `on` here is going to cause errors, because they're being bound each time submit is called.  

So if there's an error that that's fixed and re-submitted, the methods will be bound twice, causing the callbacks to fire multiple times.

To fix that you could use `once`, or `bindStoreMixin` with `asyncButton`